cli/sql: announce the target tenant in the output

knz opened the pull request:
Fixes #93129.
Epic: CRDB-23559

See individual commits for details.

knz committed:
cli/sql: warn the user when connecting to the system tenant

This change ensures that a clear warning is printed in the SQL shell
when the user connects to the system tenant and there are secondary
tenants defined.

For example:
```
 % ./cockroach sql --certs-dir=~/.cockroach-demo -d cluster:system
\#
\# Welcome to the CockroachDB SQL shell.
\[...]
\#
\#
\# ATTENTION: YOUR ARE CONNECTED TO THE SYSTEM TENANT OF A MULTI-TENANT CLUSTER.
\# PROCEED WITH CAUTION. YOU ARE RESPONSIBLE FOR ENSURING THAT YOU DO NOT
\# PERFORM ANY OPERATIONS THAT COULD DAMAGE THE CLUSTER OR OTHER TENANTS.
\#
[...]
root@localhost:26257/defaultdb>
```

Release note: None

knz committed:
cli/sql: include the tenant name in the prompt when known

This change extends the SQL prompt with the name of the tenant the
prompt is connected to, if it was specified in the URL.

For example:
```
% ./cockroach sql --certs-dir=~/.cockroach-demo -d cluster:foo
...
root@localhost:26257/foo/defaultdb>
```

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/100920)
<!-- Reviewable:end -->

rafiss commented on pkg/cli/interactive_tests/test_demo_cli_integration.tcl.disabled:
```diff
@@ -47,28 +47,37 @@ eexpect "does not exist"
 send "\\q\r"
 eexpect eof
+
+spawn $argv sql --no-line-editor -u root -d cluster:demo-tenant
```
it seems like this test file is being skipped. should we re-enable it?

there's a lot of other disabled ones too. perhaps we can check that they don't regress with this change, even if they are not re-enabled right now.

knz replied:
Good call. I'll check them manually in this PR, then send another PR separately to re-enable them.

knz committed:
cli/interactive_tests: make test_client_side_checking easier to run locally

Sometimes the shell invoked by `expect` adds additional escape
characters in-between the output of a process and the next prompt.
This patch makes the test more robust in that case.

Release note: None

knz commented:
I have checked manually that all the disabled tests still pass. I'll send a separate PR to un-skip them.

bors r=rafiss

knz commented:
lint error

bors r-

knz committed:
cli/democluster: rename "demo-tenant" to "demoapp"

We aim to de-emphasize the word "tenant" in user-facing features.
This patch achieves this for `cockroach demo`
by making the tenant name "demoapp" instead of "demo-tenant".

Release note: None

knz commented:
bors r=rafiss
